src/main/java/core/basesyntax/FileReaderImpl.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+
+public class FileReaderImpl implements FileReader {
+    @Override
+    public List<String> read(String filePath) throws IOException {
```

Comment:
You should avoid throwing the generic IOException. Instead, catch the IOException inside the method and rethrow a RuntimeException with an informative message. This way, you can provide more context if an error occurs, making it easier to debug. Use try-with-resources to ensure that resources are closed properly.